tracker/dmlc_tracker/mpi.py

```diff
@@ -0,0 +1,44 @@
+"""
+DMLC submission script, MPI version
+"""
+# pylint: disable=invalid-name
+from __future__ import absolute_import
+
+import os
+import subprocess
+from threading import Thread
+from . import tracker
+
+def submit(args):
+    """Submission script with MPI."""
+    curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
+    launcher = os.path.join(curr_path, 'launcher.py')
+
+    def mpi_submit(nworker, nserver, pass_envs):
+        """Internal closure for job submission."""
+        env = os.environ.copy()
+        for k, v in pass_envs.items():
+            env[k] = str(v)
+
+        if args.mpi_host_file is None:
+            cmd = 'mpirun -n %d' % (nworker + nserver)
+        else:
+            cmd = 'mpirun -n %d --hostfile %s ' % (nworker + nserver, args.mpi_host_file)
+
+        pass_envs['DMLC_MODE'] = args.mode
+        for k, v in pass_envs.items():
+            # for mpich2
+            if args.mode == 'mpich2':
+                cmd += ' -env %s %s' % (k, v)
+            else:
+                cmd += ' -x %s=%s ' % (k, v)
```

Comment:
but not every user know there are several mpi versions, since all provide `mpirun`